localization/ndt_scan_matcher/src/ndt_scan_matcher_core.cpp

```diff
@@ -704,6 +723,71 @@ bool NDTScanMatcher::validate_converged_param(
   return is_ok_converged_param;
 }
 
+void NDTScanMatcher::estimate_covariance(
+  const pclomp::NdtResult & ndt_result, const Eigen::Matrix4f & initial_pose_matrix,
+  std::array<double, 36> & ndt_covariance, const rclcpp::Time & sensor_ros_time)
+{
+  // Laplace approximation
+  const Eigen::Matrix2d hessian_inv = -ndt_result.hessian.inverse().block(0, 0, 2, 2);
+  const Eigen::SelfAdjointEigenSolver<Eigen::Matrix2d> lap_eigensolver(hessian_inv);
+  Eigen::Matrix2d rot = Eigen::Rotation2Dd(0.0);
+  if (lap_eigensolver.info() == Eigen::Success) {
+    const Eigen::Vector2d lap_eigen_vec = lap_eigensolver.eigenvectors().col(1);
+    const double th = std::atan2(lap_eigen_vec.y(), lap_eigen_vec.x());
+    rot = Eigen::Rotation2Dd(th);
+  }
+
+  // first result is added to mean and covariance
+  Eigen::Vector2d p2d(ndt_result.pose(0, 3), ndt_result.pose(1, 3));
+  Eigen::Vector2d tmp_centroid = p2d;
+  Eigen::Matrix2d tmp_cov = p2d * p2d.transpose();
+
+  // prepare msg
+  geometry_msgs::msg::PoseArray multi_ndt_result_msg;
+  geometry_msgs::msg::PoseArray multi_initial_pose_msg;
+  multi_ndt_result_msg.header.stamp = sensor_ros_time;
+  multi_initial_pose_msg.header.stamp = sensor_ros_time;
```

Comment:
```suggestion
  multi_initial_pose_msg.header.stamp = sensor_ros_time;
  multi_ndt_result_msg.header.frame_id = map_frame_;
  multi_initial_pose_msg.header.frame_id = map_frame_;
```
need frame_id to show RViz